ddspipe_participants/src/cpp/participant/dyn_types/DynTypesParticipant.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DynTypesParticipant.cpp
+ */
+
+#include <memory>
+
+#include <cpp_utils/Log.hpp>
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastrtps/rtps/participant/RTPSParticipant.h>
+#include <fastrtps/rtps/RTPSDomain.h>
+#include <fastrtps/types/DynamicType.h>
+#include <fastrtps/types/DynamicTypePtr.h>
+#include <fastrtps/types/TypeObjectFactory.h>
+
+#include <ddspipe_core/types/data/DynamicTypeData.hpp>
+#include <ddspipe_core/types/dynamic_types/types.hpp>
+
+#include <ddspipe_participants/reader/auxiliar/BlankReader.hpp>
+#include <ddspipe_participants/reader/rtps/SimpleReader.hpp>
+#include <ddspipe_participants/reader/rtps/SpecificQoSReader.hpp>
+#include <ddspipe_participants/writer/auxiliar/BlankWriter.hpp>
+
+#include <ddspipe_participants/participant/dyn_types/DynTypesParticipant.hpp>
+
+namespace eprosima {
+namespace ddspipe {
+namespace participants {
+
+using namespace eprosima::ddspipe::core;
+using namespace eprosima::ddspipe::core::types;
+
+DynTypesParticipant::DynTypesParticipant(
+        std::shared_ptr<SimpleParticipantConfiguration> participant_configuration,
+        std::shared_ptr<PayloadPool> payload_pool,
+        std::shared_ptr<DiscoveryDatabase> discovery_database)
+    : SimpleParticipant(
+        participant_configuration,
+        payload_pool,
+        discovery_database)
+    , type_object_reader_(std::make_shared<InternalReader>(
+                this->id(),
+                this->payload_pool_))
+{
+    // Do nothing
+}
+
+DynTypesParticipant::~DynTypesParticipant()
+{
+    dds_participant_->set_listener(nullptr);
+    eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(dds_participant_);
+
+}
+
+void DynTypesParticipant::init()
+{
+    CommonParticipant::init();
+    initialize_internal_dds_participant_();
+}
+
+std::shared_ptr<IWriter> DynTypesParticipant::create_writer(
+        const ITopic& topic)
+{
+    return std::make_shared<BlankWriter>();
+}
+
+std::shared_ptr<IReader> DynTypesParticipant::create_reader(
+        const ITopic& topic)
+{
+    // Can only create DDS Topics
+    const DdsTopic* dds_topic_ptr = dynamic_cast<const DdsTopic*>(&topic);
```

Comment:
This should be done with discriminator before. And type obejct topic should not be DdsTopic type